packages/kotti-ui/source/kotti-navbar/components/NavbarMenu.vue

```diff
@@ -30,17 +30,24 @@
 <script lang="ts">
 import { defineComponent } from '@vue/composition-api'
 
+import { makeProps } from '../../make-props'
+import { KottiNavbar } from '../types'
+
+namespace KottiNavbarMenu {
+	export const propsSchema = KottiNavbar.propsSchema.pick({
+		isNarrow: true,
+		sections: true,
+	})
+}
+
 import NavbarTooltip from './NavbarTooltip.vue'
 
 export default defineComponent({
 	name: 'KtNavbarMenu',
 	components: {
 		NavbarTooltip,
 	},
-	props: {
-		isNarrow: { default: false, type: Boolean },
-		sections: { required: true, type: Array },
-	},
+	props: makeProps(KottiNavbarMenu.propsSchema),
```

Comment:
This is not enough to deeply generate defaults for `link.component`. You need to manually parse the relevant props with `zod`. As mentioned in another comment, Vue does not allow modifying props in the validators which means that `makeProps` cannot implement this as a feature.